user-registration-app/src/app/views/registration-form/registration-form.component.ts

```diff
@@ -161,21 +161,28 @@ export class RegistrationFormComponent implements OnInit {
 
   ngOnInit() {
     this.route.data.subscribe(({ registration }) => {
-      this.registrationForm = registration;
-      this.registrationForm.firstHackathon = false;
-      this.registrationForm.eighteenBeforeEvent = null;
-      this.registrationForm.mlhcoc = null;
-      this.registrationForm.mlhdcp = null;
-      this.parsePhone(this.registrationForm.phone);
-      this.phoneNoUse = this.prettifiedPhone;
-      this.diet_restr = this.registrationForm.dietaryRestriction !== null;
+      this.registrationForm = new Registration();
+      //Data to keep
+      this.nukeData(this.registrationForm, registration)
+      
       setTimeout(() => {
         this.progress.complete();
         Materialize.updateTextFields();
       },         750);
     });
   }
 
+  nukeData(registrationForm, registration){
+    this.registrationForm.firstName = registration.firstName;
+    this.registrationForm.lastName = registration.lastName;
+    this.registrationForm.ethnicity = registration.ethnicity;
+    this.registrationForm.gender = registration.gender;
+    this.registrationForm.allergies = registration.allergies;
+    // this.registrationForm.major = registration.major;
+    // this.registrationForm.university = registration.university;
```

Comment:
Is there a reason these are commented out?

Reply:
There's a UI bug when they get filled in, because this solves a bug that's out in production right now I'd like to just not fill these in and fix the UI issue later